From c5a51e721331175a3ea3f42093c4336ad2697889 Mon Sep 17 00:00:00 2001
From: Devaraj Ranganna <devaraj.ranganna@arm.com>
Date: Tue, 21 May 2024 12:56:39 +0000
Subject: [PATCH] Enable access to mbedtls "private" fields

Define the macro `MBEDTLS_ALLOW_PRIVATE_ACCESS` so that, the Mbed TLS structure
fields that are marked as private are still accessible.

Signed-off-by: Devaraj Ranganna <devaraj.ranganna@arm.com>
---
 src/pkcs11/core_pkcs11_test.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/pkcs11/core_pkcs11_test.c b/src/pkcs11/core_pkcs11_test.c
index 71bad5a..6dcc8b5 100644
--- a/src/pkcs11/core_pkcs11_test.c
+++ b/src/pkcs11/core_pkcs11_test.c
@@ -32,6 +32,11 @@
 #include <stdlib.h>
 #include <string.h>

+/**
+ *  @brief Declaring MBEDTLS_ALLOW_PRIVATE_ACCESS allows access to mbedtls "private" fields.
+ */
+#define MBEDTLS_ALLOW_PRIVATE_ACCESS
+
 /* corePKCS11 includes. */
 #include "core_pki_utils.h"
 #include "core_pkcs11.h"
--
2.43.2

